ttnn/cpp/ttnn/operations/eltwise/binary_ng/device/binary_ng_program_factory.cpp

```diff
@@ -171,6 +184,10 @@ void set_or_update_runtime_arguments(
     F handle_args) {
     const auto& a = tensor_args.input_tensor_a;
     const auto& b = tensor_args.input_tensor_b;
+    const auto out_rank = c.logical_shape().rank();
+    auto aND = extract_nD_dims(a, out_rank);
+    auto bND = b.has_value() ? extract_nD_dims(c, out_rank) : 1;
```

Comment:
Was this meant to be
```suggestion
    auto bND = b.has_value() ? extract_nD_dims(*b, out_rank) : 1;
```

Reply:
why *b, not b?

Reply:
> Was this meant to be

yes.

Reply:
> why *b, not b?

since b is a std::optional<Tensor> unlike a and c which are Tensors

Reply:
OK, actually I meant b.value(). *b is good too. No problem.

Reply:
Oh I get what you mean.. I used to use the .value() as well until a while ago.
I remember Patrick mentioned this 
_The difference is that .value() performs a check and conditionally throws. operator-> and operator* have a precondition that a value is present. Those should be preferred when some other condition has already verified a value is present._